pkg/operator/podidentity/podidentitywebhook_controller_test.go

```diff
@@ -0,0 +1,317 @@
+/*
+Copyright 2018 The OpenShift Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package podidentity
+
+import (
+	"context"
+	"fmt"
+	clientgotesting "k8s.io/client-go/testing"
+	"strings"
+	"testing"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/golang/mock/gomock"
+	admissionv1 "k8s.io/api/admissionregistration/v1"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1 "k8s.io/api/policy/v1"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/kubernetes"
+	fakeclientgo "k8s.io/client-go/kubernetes/fake"
+	"k8s.io/client-go/kubernetes/scheme"
+	"sigs.k8s.io/controller-runtime/pkg/client/fake"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	configv1 "github.com/openshift/api/config/v1"
+	schemeutils "github.com/openshift/cloud-credential-operator/pkg/util"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resource/resourceapply"
+)
+
+func init() {
+	log.SetLevel(log.DebugLevel)
+}
+
+func TestPodIdentityWebhookController(t *testing.T) {
+	schemeutils.SetupScheme(scheme.Scheme)
+
+	getDeployment := func(c kubernetes.Interface, name, namespace string) *appsv1.Deployment {
+		deployment, err := c.AppsV1().Deployments(namespace).Get(context.TODO(), name, v1.GetOptions{})
+		if err == nil {
```

Comment:
I imagine when someone breaks this test they will want to see the content of `err` when it's not `nil`